From 976f0dead0a65f55aa9a419d7520bfa2fc40ce8a Mon Sep 17 00:00:00 2001
From: Allen Winter <winter@kde.org>
Date: Mon, 5 May 2025 08:32:40 -0400
Subject: [PATCH] todoviewsortfilterproxymodel.cpp - fix regression in
 filterAcceptsRow

In filterAcceptsRow don't worry about invalid source_parent.
Now todos are displayed again.

BUG: 473847
(cherry picked from commit 5c30c6614f3d862051fe51335e2631a7aded9d59)
---
 src/todo/todoviewsortfilterproxymodel.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/todo/todoviewsortfilterproxymodel.cpp b/src/todo/todoviewsortfilterproxymodel.cpp
index 9bce7040..37b109d0 100644
--- a/src/todo/todoviewsortfilterproxymodel.cpp
+++ b/src/todo/todoviewsortfilterproxymodel.cpp
@@ -28,10 +28,6 @@ void TodoViewSortFilterProxyModel::sort(int column, Qt::SortOrder order)
 
 bool TodoViewSortFilterProxyModel::filterAcceptsRow(int source_row, const QModelIndex &source_parent) const
 {
-    if (!source_parent.isValid()) {
-        return false;
-    }
-
     bool ret = QSortFilterProxyModel::filterAcceptsRow(source_row, source_parent);
 
     if (ret && mCalFilter) {
-- 
GitLab

